### PR TITLE
feat: add integral context to RLCR review prompts

### PR DESCRIPTION
**Pull request by zevorn**
## Summary

- Add the Integral (I) component to the RLCR feedback loop, completing a PID control model for Codex reviews
- Codex now sees accumulated commit history (since loop start) and the last 3 rounds' summaries/reviews alongside the existing goal tracker (P) and current summary (D)
- Validate `BASE_COMMIT` with `cat-file -e` before `git log` to prevent `set -e` crash on corrupted state files; annotate fallback when base commit is unavailable

## Changes

| File | Change |
|------|--------|
| `prompt-template/codex/commit-history-section.md` | New template for the I component |
| `hooks/loop-codex-stop-hook.sh` | Gather commit log, build round file refs, render template, pass to both review prompts |
| `prompt-template/codex/regular-review.md` | Insert `{{COMMIT_HISTORY_SECTION}}` between summary and Part 1 |
| `prompt-template/codex/full-alignment-review.md` | Same insertion |
| `tests/test-commit-history-section.sh` | 9 tests covering all edge cases |
| Version bump | 1.16.0 -> 1.17.0 |

## PID Model

- **P** (proportional): goal-tracker.md — current state vs acceptance criteria gap
- **I** (integral): accumulated commits + last 3 rounds' summary/review files — trajectory and patterns
- **D** (derivative): current round summary — what changed this round

## Test plan

- [x] Round 0: `(no commits yet)` and `(first round, no prior history)` ✓
- [x] Round 2+: commit log and round file references render correctly ✓
- [x] Corrupted `base_commit`: hook does not crash, falls back with `(base commit unavailable, showing recent branch commits)` annotation ✓
- [x] Missing template: fallback renders full section including commits, round files, and directive language ✓
- [x] Round 1 boundary: only round 0 referenced ✓
- [x] Empty BASE_COMMIT: triggers annotation ✓

All 9/9 tests pass (`tests/test-commit-history-section.sh`).

### CODE REVIEW

**Comment by SihaoLiu** [comment]
@codex please review this PR

File: hooks/loop-codex-stop-hook.sh [hunk 1]
```diff
@@ -965,6 +965,37 @@ COMPLETED_ITERATIONS=$((CURRENT_ROUND + 1))
 PREV_ROUND=$(( CURRENT_ROUND > 0 ? CURRENT_ROUND - 1 : 0 ))
 PREV_PREV_ROUND=$(( CURRENT_ROUND > 1 ? CURRENT_ROUND - 2 : 0 ))
 
+# Integral component: accumulated commit history and recent round references
+# Validate BASE_COMMIT is a reachable git object before using it in git log
+if [[ -n "$BASE_COMMIT" ]] && git -C "$PROJECT_ROOT" cat-file -e "$BASE_COMMIT" 2>/dev/null; then
+    COMMIT_HISTORY=$(git -C "$PROJECT_ROOT" log --oneline --no-decorate --reverse "$BASE_COMMIT"..HEAD 2>/dev/null | tail -80)
```
[the remark this answers is not in the document]
Comment (SihaoLiu) [reply hooks/loop-codex-stop-hook.sh]: @zevorn

**Review by SihaoLiu — APPROVED** [review]
lgtm
